iml-system-docker-tests/tests/docker_integration_test.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use iml_system_test_utils::{docker, get_local_server_names, iml, vagrant};
+use std::{
+    collections::{hash_map::RandomState, HashMap},
+    time::Duration,
+};
+use tokio::time::delay_for;
+
+async fn setup() -> Result<(), Box<dyn std::error::Error>> {
+    // remove the stack if it is running and clean up volumes and network
+    docker::remove_iml_stack().await?;
+    docker::system_prune().await?;
+    docker::volume_prune().await?;
+    docker::configure_docker_overrides().await?;
+    docker::stop_swarm().await?;
+    iml_systemd::restart_unit("docker.service".into()).await?;
+
+    docker::start_swarm().await?;
+    docker::set_password().await?;
+
+    let snapshot_map = vagrant::get_snapshots().await?;
+    let snapshots = snapshot_map
+        .values()
+        .next()
+        .expect("Couldn't retrieve snapshot list.");
+    
+    if snapshots.len() == 0 {
+        // Destroy any vagrant nodes that are currently running
+        vagrant::destroy().await?;
+    }
+
+    Ok(())
+}
+
+async fn run_fs_test<S: std::hash::BuildHasher>(
+    config: &vagrant::ClusterConfig,
+    docker_setup: &docker::DockerSetup,
+    server_map: HashMap<String, &[String], S>,
+    fs_type: vagrant::FsType,
+) -> Result<(), Box<dyn std::error::Error>> {
+    setup().await?;
+    docker::configure_docker_setup(&docker_setup).await?;
+
+    docker::deploy_iml_stack().await?;
+
+    vagrant::setup_deploy_docker_servers(&config, server_map).await?;
+    vagrant::create_fs(fs_type, &config.storage_servers()[..]).await?;
+
+    // Wait three seconds before detecting the filesystem
```

Comment:
```suggestion
    // Wait ten seconds before detecting the filesystem
```

Reply:
What happens if we don't wait here?

Reply:
There's a chance it will hang detecting the filesystem (doesn't happen every time).